nipype/interfaces/freesurfer/petsurfer.py

```diff
@@ -0,0 +1,584 @@
+    >>> gtmpvc.inputs.km_hb = ['11 12 50 51']
+    >>> gtmpvc.inputs.no_rescale = True
+    >>> gtmpvc.inputs.save_input = True
+    >>> gtmpvc.cmdline  # doctest: +NORMALIZE_WHITESPACE
+    'mri_gtmpvc --auto-mask 1.000000 0.100000 --default-seg-merge \
+    --i sub-01_ses-baseline_pet.nii.gz --km-hb 11 12 50 51 --km-ref 8 47 --no-rescale \
+    --psf 4.000000 --o pvc --reg sub-01_ses-baseline_pet_mean_reg.lta --save-input \
+    --seg gtmseg.mgz'
+    """
+
+    _cmd = "mri_gtmpvc"
+    input_spec = GTMPVCInputSpec
+    output_spec = GTMPVCOutputSpec
+
+    def _list_outputs(self):
+        outputs = self.output_spec().get()
+        # Get the top-level output directory
+        if not isdefined(self.inputs.pvc_dir):
+            pvcdir = os.getcwd()
+        else:
+            pvcdir = os.path.abspath(self.inputs.pvc_dir)
+        outputs["pvc_dir"] = pvcdir
+
+        # Assign the output files that always get created
+        outputs["ref_file"] = os.path.join(pvcdir, "km.ref.tac.dat")
+        outputs["hb_nifti"] = os.path.join(pvcdir, "km.hb.tac.nii.gz")
+        outputs["hb_dat"] = os.path.join(pvcdir, "km.hb.tac.dat")
+        outputs["nopvc_file"] = os.path.join(pvcdir, "nopvc.nii.gz")
+        outputs["gtm_file"] = os.path.join(pvcdir, "gtm.nii.gz")
+        outputs["gtm_stats"] = os.path.join(pvcdir, "gtm.stats.dat")
+        outputs["reg_pet2anat"] = os.path.join(pvcdir, "aux/bbpet2anat.lta")
+
+        # Assign the conditional outputs
+        if isdefined(self.inputs.save_input) and self.inputs.save_input:
+            outputs["input_file"] = os.path.join(pvcdir, "input.nii.gz")
+        if isdefined(self.inputs.save_yhat0) and self.inputs.save_yhat0:
+            outputs["yhat0"] = os.path.join(pvcdir, "yhat0.nii.gz")
+        if isdefined(self.inputs.save_yhat) and self.inputs.save_yhat:
+            outputs["yhat"] = os.path.join(pvcdir, "yhat.nii.gz")
+        if isdefined(self.inputs.save_yhat_full_fov) and self.inputs.save_yhat_full_fov:
+            outputs["yhat_full_fov"] = os.path.join(pvcdir, "yhat_full_fov.nii.gz")
+        if (
+            isdefined(self.inputs.save_yhat_with_noise)
+            and self.inputs.save_yhat_with_noise
+        ):
+            outputs["yhat_with_noise"] = os.path.join(pvcdir, "yhat_with_noise.nii.gz")
+        if isdefined(self.inputs.mgx) and self.inputs.mgx:
+            outputs["mgx_ctxgm"] = os.path.join(pvcdir, "mgx.ctxgm.nii.gz")
+            outputs["mgx_subctxgm"] = os.path.join(pvcdir, "mgx.subctxgm.nii.gz")
+            outputs["mgx_gm"] = os.path.join(pvcdir, "mgx.gm.nii.gz")
+        if isdefined(self.inputs.rbv) and self.inputs.rbv:
+            outputs["rbv"] = os.path.join(pvcdir, "rbv.nii.gz")
+        if isdefined(self.inputs.opt) and self.inputs.opt:
+            outputs["opt_params"] = os.path.join(pvcdir, "aux/opt.params.dat")
+
+        return outputs
+
+    def _format_arg(self, name, spec, value):
+        return super(GTMPVC, self)._format_arg(name, spec, value)
```

Comment:
is this needed? seems like its just calling the inherited method

Reply:
Yes. This seems to be needed. Otherwise I get this message: "import nipype.interfaces.freesurfer as fs
[autoreload of nipype.interfaces.freesurfer.petsurfer failed: Traceback (most recent call last):
  File "/Users/martinnorgaard/opt/anaconda3/lib/python3.9/site-packages/IPython/extensions/autoreload.py", line 245, in check
    superreload(m, reload, self.old_objects)
  File "/Users/martinnorgaard/opt/anaconda3/lib/python3.9/site-packages/IPython/extensions/autoreload.py", line 410, in superreload
    update_generic(old_obj, new_obj)
  File "/Users/martinnorgaard/opt/anaconda3/lib/python3.9/site-packages/IPython/extensions/autoreload.py", line 347, in update_generic
    update(a, b)
  File "/Users/martinnorgaard/opt/anaconda3/lib/python3.9/site-packages/IPython/extensions/autoreload.py", line 302, in update_class
    if update_generic(old_obj, new_obj): continue
  File "/Users/martinnorgaard/opt/anaconda3/lib/python3.9/site-packages/IPython/extensions/autoreload.py", line 347, in update_generic
    update(a, b)
  File "/Users/martinnorgaard/opt/anaconda3/lib/python3.9/site-packages/IPython/extensions/autoreload.py", line 266, in update_function
    setattr(old, name, getattr(new, name))
ValueError: _format_arg() requires a code object with 1 free vars, not 0
]"

Reply:
That looks like an IPython autoreload problem. I would try deleting it and see whether it causes any real problems.